playbooks/common/openshift-cluster/upgrades/pre/verify_health_checks.yml

```diff
@@ -0,0 +1,13 @@
+---
+- name: Verify Host Requirements
+  hosts: oo_all_hosts
+  roles:
+  - openshift_health_checker
+  vars:
+  - r_openshift_health_checker_playbook_context: "upgrade"
+  post_tasks:
+  - action: openshift_health_check
+    args:
+      checks:
+      - disk_availability
```

Comment:
If I understand https://docs.openshift.org/latest/install_config/upgrading/blue_green_deployments.html#blue-green-deployments-preparing-for-upgrade correctly, it looks like in the case of a `blue-green` upgrade, an `in-place` upgrade is still run on existing `masters` and `etcd` hosts, but not on existing `nodes`. Instead, existing `nodes` are labeled with `color=blue` (or whatever value a user decides on). Then, [entirely new nodes are created with the new version of OpenShift using a playbook](https://docs.openshift.org/latest/install_config/upgrading/blue_green_deployments.html#blue-green-creating-labeling-green-nodes). Once green nodes are ready, pods are evacuated from blue nodes to green.

Therefore, I think that if the upgrade `context` is set, and the host is in the `nodes` group, it is safe to assume that it is going through an `in-place` upgrade (since a blue-green upgrade would be creating entirely new nodes instead). `masters` and `etcd` hosts can be treated as `in-place` for both cases.